frontend/app/controllers/Event.scala

```diff
@@ -134,13 +134,12 @@ class Event(
   }
 
   def buy(id: String): Action[AnyContent] = eventbriteService.getEvent(id) match {
-      case Some(event@(_: GuLiveEvent)) =>
-        BuyAction(id).async { implicit request =>
-          if (event.isBookableByTier(request.subscriber.subscription.plan.tier))
-            redirectSignedInMemberToEventbrite(event)
-          else suggestUserUpgrades
-        }
-      case Some(event@(_: MasterclassEvent)) =>
+      // One conditional now covers Live Events AND Masterclasses...
+
+      // Logged out readers and non-members will go via the onUnauthenticated route
+      // They, and any ineligible-tier Members will get a "SOLD OUT" EventBrite page if the event has membership-only tickets.
```

Comment:
where is the logic which checks for ineligible tier happening?

Reply:
Technically, it's in the comment of the description field on the eventbrite page. See the 'ticketing' val in Eventbrite.scala

Reply:
And TicketSaleDates.scala